### PR TITLE
PA-1582 Property Detail View - 

### PR DESCRIPTION
**Pull request by devinleighsmith**
Auto-populate form with parcel layer data in the following situations:
1) There is a link on the popup that is opened when clicking on the parcel layer. 
2) If either the pid or pin is entered, the parcel data layer may populate data if there is a match.
3) If the civil address is entered, the parcel data layer may populate based on the returned lat/lng from geocoder.

Also Corrected nearly all of the tests warnings.

### CODE REVIEW

File: frontend/src/assets/images/draft-parcel-icon.svg [hunk 1]
```diff
@@ -1 +1 @@
-<svg version="1.2" xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink" overflow="visible" preserveAspectRatio="none" viewBox="0 0 24 24" width="40" height="40"><g><path xmlnsDefault="http://www.w3.org/2000/svg" d="M18 8c0-3.31-2.69-6-6-6S6 4.69 6 8c0 4.5 6 11 6 11s6-6.5 6-11zm-8 0c0-1.1.9-2 2-2s2 .9 2 2-.89 2-2 2c-1.1 0-2-.9-2-2zM5 20v2h14v-2H5z" style="fill: rgb(46, 133, 64);" vector-effect="non-scaling-stroke"/></g></svg>
```
Comment (Fosol) [inline frontend/src/assets/images/draft-parcel-icon.svg:1]: Curious what the reason for removing?

File: frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx [hunk 2]
```diff
@@ -1,33 +1,108 @@
...
+  /**
+   * function to find GeoJSON shape matching the passed non-zero padded pid.
+   * @param pid
+   */
+  findByPid: (pid: string) => Promise<FeatureCollection>;
+  /**
+   * function to find GeoJSON shape matching the passed pin.
+   * @param pin
+   */
+  findByPin: (pin: string) => Promise<FeatureCollection>;
```
Comment (Fosol) [inline frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx:28]: I don't think the layer has PIN values.

File: frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx [hunk 3]
```diff
@@ -1,33 +1,108 @@
...
 export const useLayerQuery = (url: string, geometryName: string = 'SHAPE'): IUserLayerQuery => {
+  const parcelLayerData = useSelector<RootState, IParcelLayerData | null>(
+    state => state.parcelLayerData?.parcelLayerData,
+  );
   const findOneWhereContains = useCallback(
     async (latlng: LatLng): Promise<FeatureCollection> => {
       const data: FeatureCollection = (
         await axios.get(
-          `${url}&count=1&cql_filter=CONTAINS(${geometryName}, SRID=4326;POINT ( ${latlng.lng} ${latlng.lat}))`,
+          `${url}&srsName=EPSG:4326&count=1&cql_filter=CONTAINS(${geometryName},SRID=4326;POINT ( ${latlng.lng} ${latlng.lat}))`,
```
Comment (Fosol) [inline frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx:75]: Looks like possible duplication of params that might benefit from extracting to a shared variable or const.

File: frontend/src/features/mapSideBar/hooks/useQueryParamSideBar.tsx [hunk 4]
```diff
@@ -21,14 +22,20 @@ const useQueryParamSideBar = (): IMapSideBar => {
   const history = useHistory();
 
   const searchParams = queryString.parse(location.search);
-  useEffect(() => {
+  useDeepCompareEffect(() => {
```
Comment (Fosol) [inline frontend/src/features/mapSideBar/hooks/useQueryParamSideBar.tsx:25]: Why the change here?

File: frontend/src/features/properties/components/forms/subforms/PidPinForm.tsx [hunk 5]
```diff
@@ -44,7 +57,12 @@ const PidPinForm: FunctionComponent<PidPinProps> = (props: PidPinProps) => {
           tooltip={PidTooltip}
           disabled={props.disabled}
           pattern={RegExp(/^[\d\- ]*$/)}
-          onBlurFormatter={(pid: string) => pid.replace(pid, pidFormatter(pid))}
+          onBlurFormatter={(pid: string) => {
+            if (pid && getIn(touched, withNameSpace('pid'))) {
+              debouncedHandlePidChange(pid);
```
Comment (Fosol) [inline frontend/src/features/properties/components/forms/subforms/PidPinForm.tsx:62]: Does this remove the `-` dashes when making a request to the layer?

File: frontend/src/reducers/parcelLayerDataSlice.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+export const saveParcelLayerData = createAction<IParcelLayerData>('saveParcelLayerData');
+export const clearParcelLayerData = createAction('clearParcelLayerData');
+/**
+ * The following is a shorthand method for creating a reducer with paired actions and action creators.
+ * All functionality related to this concept is contained within this file.
+ * See https://redux-toolkit.js.org/api/createslice for more details.
+ */
+const parcelLayerDataSlice = createSlice({
```
Comment (Fosol) [inline frontend/src/reducers/parcelLayerDataSlice.ts:15]: Happy to see the toolkit being used.

**Review by Fosol — APPROVED** [review]
Thanks for updating the tests.  Looking forward to testing.

**Review by jdtoombs — APPROVED** [review]

File: frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx (hunk 2 above)
Comment (devinleighsmith) [reply frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx:28]: it does.

File: frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx (hunk 3 above)
Comment (devinleighsmith) [reply frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx:75]: fair

File: frontend/src/features/mapSideBar/hooks/useQueryParamSideBar.tsx (hunk 4 above)
Comment (devinleighsmith) [reply frontend/src/features/mapSideBar/hooks/useQueryParamSideBar.tsx:25]: useEffect wasn't picking up changes within the object dependency array since the reference was the same.

File: frontend/src/assets/images/draft-parcel-icon.svg (hunk 1 above)
Comment (devinleighsmith) [reply frontend/src/assets/images/draft-parcel-icon.svg:1]: Umm... I'll look into this, I corrected an svg but don't remember removing one.

File: frontend/src/features/properties/components/forms/subforms/PidPinForm.tsx (hunk 5 above)
Comment (devinleighsmith) [reply frontend/src/features/properties/components/forms/subforms/PidPinForm.tsx:62]: It does at the container level via .replace(/-/g, '')
